api/handlers/products.js

```diff
@@ -0,0 +1,276 @@
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 200,
+            body: JSON.stringify(result.rows)
+        }
+    } else{
+       
+            const result = await pool.query('SELECT * FROM products ORDER BY id ASC')
+            return {
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                    
+                  },
+                statusCode: 200,
+                body: JSON.stringify(result.rows)
+            }
+    }
+
+} catch(error){
+    return {
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+           
+          },
+        statusCode: 500,
+        body: JSON.stringify({error: 'Internal server error'})
+    }
+
+}
+     
+}
+//get categories
+module.exports.getCategories = async (event) => {
+   
+    try{
+        const result =  await pool.query('SELECT DISTINCT category FROM products');
+        return{
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+                
+              },
+            statusCode: 200,
+            body: JSON.stringify(result.rows)
+        }
+    } catch(error){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 500,
+            body: JSON.stringify({error: 'Internal server error'})
+        }
+    }
+}
+//get product by ID
+module.exports.getProductById = async (event) => {
+    try{
+        const id = parseInt(event.pathParameters.id)
+        //if id is not valid return 400 and invalid ID
+        if(isNaN(id)){
+            return {
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                   
+                  },
+                statusCode: 400,
+                body: JSON.stringify({message: 'Invalid ID'}),
+            };
+        }
+        const result = await pool.query('SELECT * FROM products WHERE id=$1', [id]);
+        if(result.rows.length === 0){
+            return {
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                   
+                  },
+                statusCode: 404,
+                body: JSON.stringify({ error: "Product not found" })
+            
+        }
+    }
+    else{
+        return {
+            
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 200,
+            body: JSON.stringify(result.rows)
+        }
+    }
+    }catch(error){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 500,
+            body: JSON.stringify({error: 'Internal server error'})
+        }
+    }
+}
+//create new prdouct
+module.exports.post = async (event) => {
```

Comment:
The create new product and update product APIs don't have any security checks on them so anyone could change the products on the site, making them free or making up completely fake products!

There should be a check that the person making this API call is a registered user, and that they have permission to update the products — like they have an administrator role, or something like that.